File: src/gestorAplicacion/Entrega.java
package gestorAplicacion;
import java.io.Serializable;
import java.util.ArrayList;
import gestorAplicacion.personas.Paciente;
import gestorAplicacion.Pago;


public class Entrega implements Serializable{
    private static final long serialVersionUID = 1L;
	private int id;
    private Paciente paciente;
    private String domicilio;
    private Medicamentos medicamentos;
    private estadoEntrega estado;
    public static ArrayList<Entrega> lista= new ArrayList<Entrega>();
    private Pago pago;
    
    //constructor
    public Entrega(int id, Paciente paciente, String domicilio, Medicamentos medicamentos, estadoEntrega estado) {
        this.id = id;
        this.paciente = paciente;
        this.domicilio = domicilio;
        this.medicamentos = medicamentos;
        this.estado = estado;
        lista.add(this);
    }
    
    public Entrega(Paciente paciente, String domicilio, Medicamentos medicamentos, estadoEntrega estado) {
        this.id = (int)(Math.random()*10000+1);
        this.paciente = paciente;
        this.domicilio = domicilio;
        this.medicamentos = medicamentos;
        this.estado = estado;
        lista.add(this);
    }
    
    //getter & setter
    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public Paciente getPaciente() {
        return paciente;
    }
    public void setPaciente(Paciente paciente) {
        this.paciente = paciente;
    }
    public String getDomicilio() {
        return domicilio;
    }
    public void setDomicilio(String domicilio) {
        this.domicilio = domicilio;
    }
    public Medicamentos getMedicamentos() {
        return medicamentos;
    }
    public void setMedicamentos(Medicamentos medicamentos) {
        this.medicamentos = medicamentos;
    }
    public estadoEntrega getEstado() {
        return estado;
    }
    public void setEstado(estadoEntrega estado) {
        this.estado = estado;
    }

    public Pago getPago() {
        return pago;
    }

    public void setPago(Pago pago) {
        this.pago = pago;
    }

    
    //metodos
    public static void crearEntrega(int id, Paciente paciente,String domicilio, Medicamentos medicamentos,estadoEntrega estado){
        new Entrega (id, paciente, domicilio, medicamentos, estado);
        
    }
    
    public static int informeEntregas(estadoEntrega solicitado){// se ingresa el tipo de entrega y se busca en la lista cuantas hay
        int eeSolicitada=0;
        for (Entrega ee : lista) {
            if(ee.getEstado()==solicitado){
                eeSolicitada++;
            }
        }
        return eeSolicitada;
    }
    
    
}

File: Python/src/gestorAplicacion/Entrega.py

import random as rd
class Entrega():
    lista=[]
    # constructor
    def __init__(self, id, paciente, domicilio, medicamentos, estado):
        self.id = ((rd.random() * 10000 + 1))
        self.paciente = paciente
        self.domicilio = domicilio
        self.medicamentos = medicamentos
        self.estado = estado
        self.lista.add(self)

    # getter & setter
    def getId(self):
        return self.id

    def setId(self, id):
        self.id = id

    def getPaciente(self):
        return self.paciente

    def setPaciente(self, paciente):
        self.paciente = paciente

    def getDomicilio(self):
        return self.domicilio

    def setDomicilio(self, domicilio):
        self.domicilio = domicilio

    def getMedicamentos(self):
        return self.medicamentos

    def setMedicamentos(self, medicamentos):
        self.medicamentos = medicamentos

    def getEstado(self):
        return self.estado

    def setEstado(self, estado):
        self.estado = estado

    def getPago(self):
        return self.pago

    def setPago(self, pago):
        self.pago = pago

    # metodos
    @classmethod
    def crearEntrega(cls, id, paciente, domicilio, medicamentos, estado):
        Entrega(id, paciente, domicilio, medicamentos, estado)

    @classmethod
    def informeEntregas(cls, solicitado):
        #  se ingresa el tipo de entrega y se busca en la lista cuantas hay
        eeSolicitada = 0
        for ee in Entrega.lista :
            if ee.getEstado() == solicitado:
                eeSolicitada += 1
        return eeSolicitada